Source/JavaScriptCore/bytecode/InlineCacheCompiler.cpp

```diff
@@ -2855,19 +2855,14 @@ void InlineCacheCompiler::generateImpl(unsigned index, AccessCase& accessCase)
 
             setSpillStateForJSCall(spillState);
 
-            m_callLinkInfos[index] = makeUnique<OptimizingCallLinkInfo>(m_stubInfo->codeOrigin, codeBlock->useDataIC() ? CallLinkInfo::UseDataIC::Yes : CallLinkInfo::UseDataIC::No, nullptr);
+            CallLinkInfo::UseDataIC useDataIC = CallLinkInfo::UseDataIC::Yes;
+#if USE(JSVALUE64)
+            if (!codeBlock->useDataIC())
+                useDataIC = CallLinkInfo::UseDataIC::No;
+#endif
```

Comment:
nit: I feel like this would be clearer as

```
#if USE(JSVALUE64)
     CallLinkInfo::UseDataIC useDataIC = codeBlock->useDataIC() ? CallLinkInfo::UseDataIC::Yes : CallLinkInfo::UseDataIC::No;
#else
     CallLinkInfo::UseDataIC useDataIC = CallLinkInfo::UseDataIC::No;
#endif
```

Reply:
Why are we only disabling dataIC for JSValue64? I'm confused by this code.

Reply:
It seems that 32bit non-DataIC is not working correclty. But anyway, we can just keep it broken.